### PR TITLE
Add support for individual model constraints

### PR DESCRIPTION
**Pull request by pskiran1**

### CODE REVIEW

File: tests/test_constraint_manager.py [hunk 1]
```diff
@@ -90,11 +90,11 @@ def test_single_model_with_both_constraints(self):
                              'max': 100
```
Comment (nv-braf) [inline tests/test_constraint_manager.py:90]: Please update copyright date

File: tests/test_constraint_manager.py [hunk 2]
```diff
@@ -379,21 +591,48 @@ def _create_multi_model_both_constraints(self):
...
+                  min: 100
                 gpu_used_memory:
-                  max: 100
+                  max: 1000
+        """)
+        config = evaluate_mock_config(args, yaml_str, subcommand="profile")
+
+        return config
+
+    def _create_multi_model_with_different_all_constraints(self):
```
Comment (nv-braf) [inline tests/test_constraint_manager.py:621]: change to with_different_global_constrants

File: tests/test_constraint_manager.py [hunk 3]
```diff
@@ -379,21 +591,48 @@ def _create_multi_model_both_constraints(self):
                 constraints:
                   perf_throughput:
                     min: 100
-              model_C:
+        """)
+        config = evaluate_mock_config(args, yaml_str, subcommand="profile")
+
+        return config
+
+    def _create_multi_model_with_global_constraints(self):
```
Comment (nv-braf) [inline tests/test_constraint_manager.py:599]: change to with_matching_global_constraints

File: tests/test_constraint_manager.py [hunk 4]
```diff
@@ -367,7 +579,7 @@ def _create_single_model_both_constraints(self):
 
         return config
 
-    def _create_multi_model_both_constraints(self):
+    def _create_multi_model_with_different_model_constraints(self):
```
Comment (nv-braf) [inline tests/test_constraint_manager.py:582]: change to with_no_global_constraints

**Review by nv-braf — COMMENTED** [review]
Looks really good. Very thorough!

File: tests/test_constraint_manager.py (hunk 2 above)
Comment (pskiran1) [reply tests/test_constraint_manager.py]: Done 👍

File: tests/test_constraint_manager.py (hunk 3 above)
Comment (pskiran1) [reply tests/test_constraint_manager.py]: Done 👍

File: tests/test_constraint_manager.py (hunk 4 above)
Comment (pskiran1) [reply tests/test_constraint_manager.py]: Done 👍

File: tests/test_constraint_manager.py (hunk 1 above)
Comment (pskiran1) [reply tests/test_constraint_manager.py:90]: Done 👍

File: tests/test_constraint_manager.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (nv-braf) [inline tests/test_constraint_manager.py:1]: This should be 2022-2023

Comment (pskiran1) [reply tests/test_constraint_manager.py]: Done 👍

**Review by nv-braf — APPROVED** [review]

**Pull request merged**